sklearn/tests/test_calibration.py

```diff
@@ -410,10 +412,9 @@ def test_calibration_curve():
     assert_almost_equal(prob_true, prob_true_unnormalized)
     assert_almost_equal(prob_pred, prob_pred_unnormalized)
 
-    # probabilities outside [0, 1] should not be accepted when normalize
-    # is set to False
+    # Probabilities outside [0, 1] should not be accepted at all.
     with pytest.raises(ValueError):
-        calibration_curve([1.1], [-0.1], normalize=False)
+        calibration_curve([1], [-0.1])
```

Comment:
Could you isolate these specific 2 lines in a single test and name it `test_calibration_curve_with_unormalized_proba`.

Reply:
I am a little bit confused by this suggestion. If I understand correctly, the goal is to separate any tests that cover the normalize parameter (i.e. FutureWarnings are raised for True or False, and that `normalize` == True behaves as expected) into their own test called `test_calibration_curve_with_unnormalized_proba`.

If I have understood this goal correctly, these lines should remain in the body of the `test_calibration_curve` function because we will want to ensure that probabilities passed to `calibration_curve` are valid (i.e. between 0 and 1) even after the normalize parameter and its associated tests are removed in 1.3

Please let me know if I have misunderstood your suggestion.

Reply:
Yes the change is exactly what I had in mind:

- keep the check about raising an error
- put the code outside when we play with normalize

Reply:
Perfect, thank you kindly for the clarification!